### PR TITLE
feat: Add ability to support spot options for launch template

### PR DESCRIPTION
**Pull request by davidweizheng**
# PR o'clock

## Description

Add ability to support spot options for launch template to address this issue: https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1626

### Checklist

- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by sergeyshevch** [comment]
Maybe someone can review and accept it? I also interested in such feature

**Comment by bryantbiggs** [comment]
@davidweizheng could you update an example so that it can be tested/validated and also for demonstrating its use to users - please 🙏🏽  and thank you!

File: docs/spot-instances.md [hunk 1]
```diff
@@ -83,6 +83,29 @@ Launch Template support is a recent addition to both AWS and this module. It mig
...
+```hcl
+  worker_groups_launch_template = [
+    {
+      name                             = "spot-1"
+      instance_type                    = "t3.large"
+      market_type                      = "spot"
+      spot_options                     = {
+        block_duration_minutes         = "60"
+        instance_interruption_behavior = "terminate"
+        max_price                      = null
```
Comment (daroga0002) [inline docs/spot-instances.md:97]: code should be able to take a default provider values in cae of missing value for field. As a lot of them are optional we cannot require to put `null` from users. 

Here you can take a look of some constructions which we using to make such optional inputs:
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/56e93d77de58f311f1d1d7051f40bf77e7b03524/modules/node_groups/launch_template.tf#L88-L92

https://github.com/terraform-aws-modules/terraform-aws-eks/blob/56e93d77de58f311f1d1d7051f40bf77e7b03524/modules/fargate/main.tf#L66-L69

**Review by daroga0002 — CHANGES_REQUESTED** [review]

**Comment by davidweizheng** [comment]
> @davidweizheng could you update an example so that it can be tested/validated and also for demonstrating its use to users - please 🙏🏽 and thank you!

I do have example in `docs/spot-instances.md`, but I'll try to add that as an actual example. Thanks.

**Comment by sergeyshevch** [comment]
Ping this one more time )

File: docs/spot-instances.md (hunk 1 above)
Comment (davidweizheng) [reply docs/spot-instances.md]: Fixed.

**Comment by davidweizheng** [comment]
Could someone review this?

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.0.0 :tada:

**Pull request closed without merging**